src/loader/init.c

```diff
@@ -0,0 +1,149 @@
+#include <postgres.h>
+#include <pg_config.h>
+#include <access/xact.h>
+#include <commands/extension.h>
+#include <miscadmin.h>
+#include <utils/guc.h>
+#include <utils/inval.h>
+#include <parser/analyze.h>
+#include <nodes/print.h>
+
+
+#include "extension.h"
+
+/*
+ * Some notes on design:
+ *
+ * We do not check for the installation of the extension upon loading the extension and instead rely on a hook for two reasons:
+ * 1) We probably can't
+ *	- The shared_preload_libraries is called in PostmasterMain which is way before InitPostgres is called.
+ *			(Note: This happens even before the fork of the backend)
+ *	-- This means we cannot query for the existance of the extension yet because the caches are initialized in InitPostgres.
+ * 2) We actually don't want to load the extension in two cases:
+ *	  a) We are upgrading the extension.
+ *	  b) We set the guc timescaledb.disable_load.
+ *
+ *
+ */
+
+#define MIN_SUPPORTED_VERSION_STR "9.6"
+#define MIN_SUPPORTED_VERSION_NUM 90600
+
+#if PG_VERSION_NUM < MIN_SUPPORTED_VERSION_NUM
+#error "Unsupported version of PostgreSQL. Check src/init.c for supported versions."
+#endif
+
+#ifdef PG_MODULE_MAGIC
+PG_MODULE_MAGIC;
+#endif
+
+#define GUC_DISABLE_LOAD_NAME "timescaledb.disable_load"
```

Comment:
Is this GUC actually used for something, or just here in case someone wants to use it? I don't see it ever being set to `true` in the code. I think we should be very restrictive of adding GUCs unless really motivated. I'd rather avoid inflating the number of GUCs and then add it later if it turns out we really need it.

Reply:
This is a GUC i've wanted several times for debugging with clients especially if there are weird errors. It's never used in the code. The purpose is to have this for debugging. Having said this, I don't feel too strongly and will remove it if you think this is the wrong approach.